cachito/workers/pkg_managers/npm.py

```diff
@@ -27,6 +29,338 @@
+            raise ValidationError(
+                (
+                    f"lockfileVersion {lockfile_version} from {lockfile_path} is not supported "
+                    "Please use a supported lockfileVersion, which are versions 1, 2, and 3"
+                )
+            )
+
+        return cls(lockfile_path, lockfile_data)
+
+    def _get_packages(self) -> list[Package]:
+        """Return a flat list of Packages from a v2+ package-lock.json file.
+
+        Use the "packages" key in the lockfile to create a list of Package objects.
+        """
+
+        def get_package_name_from_path(package_path: str) -> str:
+            """Get the package name from the path in v2+ package-lock.json file."""
+            path = Path(package_path)
+            parent_name = Path(package_path).parents[0].name
+            is_scoped = parent_name.startswith("@")
+            return (Path(parent_name) / path.name).as_posix() if is_scoped else path.name
+
+        # Create all of the Package objects
+        paths_to_packages: dict[str, Package] = {}
+        for package_path, package_data in self._lockfile_data.get("packages", {}).items():
+            # The path of the package in the lockfile may have a different name than the one in
+            # the package data. This could be due to aliasing, etc., so keep both names handy
+            package_name = package_data.get("name") or get_package_name_from_path(package_path)
+            alias = (
+                get_package_name_from_path(package_path)
+                if "name" in package_data and package_path != ""
+                else None
+            )
+            paths_to_packages[package_path] = Package(
+                package_name, package_data, alias=alias, path=package_path
+            )
+
+        # For each Package object, we need to determine all of the Packages that depend on them
+        self._get_dependent_packages(paths_to_packages)
+        # Once we've mapped all of the dependent Packages, we no longer need the root Package
+        paths_to_packages.pop("", None)
+
+        # Remove link Packages since there is already a Package for the link target
+        return [package for package in paths_to_packages.values() if not package.is_link]
+
+    def _get_dependent_packages(self, paths_to_packages: dict[str, Package]) -> None:
+        """Determine and set all of the dependent Packages for each Package."""
+        for package_path, package in paths_to_packages.items():
+            for dependency_name in package.get_dependency_names():
+                log.debug(f"The package located at {package_path} depends on {dependency_name}")
+                for dependency_path in _get_resolvable_dep_paths(package_path, dependency_name):
+                    if dependency_path in paths_to_packages:
+                        log.debug(
+                            (
+                                f"The package located at {package_path} resolves "
```

Comment:
This log looks funny when it's about the main package (which has " " as path):

```
The package located at  depends on mocha
``` 

I do wonder if these logs here are helpful at all, and they usually add quite a lot of lines to the log output.

Reply:
Agreed. They were useful while I was debugging and I assume for review purposes, but I think we can remove before merging